azurelinuxagent/common/protocol/wire.py

```diff
@@ -961,7 +961,7 @@ def send_request_using_appropriate_channel(self, direct_func, host_func):
                              log_event=True)
                 raise
 
-        if not HostPluginProtocol.is_default_channel():
+        if ret and not HostPluginProtocol.is_default_channel():
```

Comment:
some of the lambdas passed to send_request_using_appropriate_channel do not have an explicit return; since you are using the return value, could add an explicit return to all the lambdas?

Reply:
I inspected the lambdas and couldn't find any which doesn't return. Which one are you referring to?
The lambdas in question are `azurelinuxagent.common.protocol.wire.WireClient.stream` and `azurelinuxagent.common.protocol.wire.WireClient.fetch`.

Reply:
e.g. direct_func, host_func in
```
def download_ext_handler_pkg(self, uri, destination, headers=None, use_proxy=True):
        direct_func = lambda: self.client.stream(uri, destination, headers=None, use_proxy=True)
        # NOTE: the host_func may be called after refreshing the goal state, be careful about any goal state data
        # in the lambda.
        host_func = lambda: self._download_ext_handler_pkg_through_host(uri, destination)

        try:
            success = self.client.send_request_using_appropriate_channel(direct_func, host_func)
        except Exception:
            success = False

        return success
```

Reply:
never mind, dunno what i was thinking, you are ok

Reply:
Maybe fasttrack changes are seeping in :)